### PR TITLE
[binder] End-to-end callback benchmark app

### PR DESCRIPTION
**Pull request by TaWeiTu**
Benchmark output can be retrieved with `adb logcat '*:S Benchmark:I'`

Depends on https://github.com/grpc/grpc/pull/27257.

### CODE REVIEW

**Comment by sifmelcara** [comment]
1. I think we want completely different app name and package name
2. Should let people who worked on gRPC benchmark take a quick look

**Comment by TaWeiTu** [comment]
> 1. I think we want completely different app name and package name

Will do.

> 2. Should let people who worked on gRPC benchmark take a quick look

That'd be nice. However, the benchmark part is almost the same as the one in `callback_unary_ping_pong.h` and `callback_streaming_ping_pong.h`.

**Comment by sifmelcara** [comment]
Yeah I'm just wondering if they have some different idea than creating 2 APK, install on real device, and run the test

**Comment by TaWeiTu** [comment]
> 1. I think we want completely different app name and package name

Done. Renamed to `io.grpc.binder.cpp.benchmarks.client` and `io.grpc.binder.cpp.benchmarks.server`.

File: test/cpp/microbenchmarks/bm_binder/java/io/grpc/binder/cpp/benchmarks/client/native.cc [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  static bool first = true;
+  if (first) {
+    first = false;
+    grpc::experimental::BindToOnDeviceServerService(
+        env, application, "io.grpc.binder.cpp.benchmarks.server",
+        "io.grpc.binder.cpp.benchmarks.server.ExportedEndpointService");
+    return env->NewStringUTF("Clicked 1 time");
+  } else {
+    grpc::testing::RunCallbackPingPongBinderBenchmarks(env, application);
+    return env->NewStringUTF("Clicked more than 1 time. Status not ok");
```
Comment (sifmelcara) [inline test/cpp/microbenchmarks/bm_binder/java/io/grpc/binder/cpp/benchmarks/client/native.cc:32]: Probably say something like "results are ready in logcat"

**Review by sifmelcara — COMMENTED** [review]
1. Consider separate boring new Android app code (excluding actual benchmark code) into a separate PR, which can be merged first before flow control PRs are ready
2. I think we will want to build the app in CI (`tools/internal_ci/linux/grpc_binder_transport_apk_build_in_docker.sh`)

File: test/cpp/microbenchmarks/bm_binder/java/io/grpc/binder/cpp/benchmarks/client/native.cc (hunk 1 above)
Comment (TaWeiTu) [reply test/cpp/microbenchmarks/bm_binder/java/io/grpc/binder/cpp/benchmarks/client/native.cc]: Done.

File: test/cpp/microbenchmarks/bm_binder/java/io/grpc/binder/cpp/benchmarks/server/BUILD [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    srcs = ["native.cc"],
+    linkopts = [
+        "-ldl",
+        "-llog",
+        "-lm",
+        "-lbinder_ndk",
+        "-Wl,--no-undefined",
+    ],
+    deps = [
+        # Temporarily directly depend on this target before we expose a public API
```
[the remark this answers is not in the document]
Comment (TaWeiTu) [reply test/cpp/microbenchmarks/bm_binder/java/io/grpc/binder/cpp/benchmarks/server/BUILD]: Done.

**Pull request closed without merging**